runtime/src/common/quantity.rs

```diff
@@ -15,6 +18,36 @@ impl From<u64> for Quantity {
     }
 }
 
+impl<'a> Add<&'a Quantity> for Quantity {
+    type Output = Quantity;
+
+    fn add(mut self, other: &Quantity) -> Quantity {
```

Comment:
Maybe i'm missing something, but shouldn't `Add`  not update `self`?

Reply:
So it is not `&mut self` but `mut self` (as required by the `Add` trait) which means that self is moved into the method. This means that there are no other references to it anywhere so it is safe to reuse it.

Reply:
Ah i see. Thanks :+1: